riscv/src/runtime.rs

```diff
@@ -148,6 +148,48 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
+    pub fn with_keccak(mut self) -> Self {
+        self.add_submachine(
+            "test_data::asm::keccakf::KeccakF",
+            None,
+            "keccakf",
+            [format!(
+                "instr keccakf ~ keccakf.keccakf {};", // operation
+                instr_register_params(0, 25, 25) // 25 registers for the states, covered by 26 registers of riscv
```

Comment:
It should be just 2 numbers here, the memory addresses of the input and output. The machine will use memory directly.

Reply:
I see. I can add an output pointer parameter as well. I just wonder if it'll be easier to actually just use one pointer for both input and output, as this compression/mixing function is run multiple times iteratively on the output of itself during a single keccak run.

Reply:
AFAIU, we'll need a separate output pointer due to the workings of the memory machine (to allow reading/writing in the same "row")

Reply:
Done.